### DISCUSSION TITLE
Select form field with default() does not give expected value when model $attributes is set

### DISCUSSION THREAD

**Discussion by ekateiva** [laravel/nova-issues#6573] [Ideas & Feature Requests]
- Laravel Version: 11.29.0
- Nova Version: 4.35.4
- PHP Version: 8.3.9

### Description:

I have a field like:
```
Select::make('Responsible party', 'responsible_party')
    ->options([
        'logistics' => 'Logistics company',
        'platform' => 'Our platform',
        'supplier' => 'Supplier',
 ])
    ->default('supplier'),
```

<img width="529" alt="image" src="https://github.com/user-attachments/assets/6694d5b4-dabe-447a-b146-e31b3018ad65">

The default expected value would be "Supplier" select in the field. However, I see "Our platform", because:

In the model we have:
```
protected $attributes = [
    'responsible_party' => 'platform',
];
```

I would think that `default()` in Nova context should override the default value of Laravel context.

In my opinion it is a bug or at least not expected behaviour.

Now we need to use something like `->withMeta(['value' => 'supplier'])` to get arround this behaviour.

### Detailed steps to reproduce the issue on a fresh Nova installation:

An example given above, resource with a single field and a model with $attributes.

**Comment by crynobone** [comment]
Model will always have the higher priority over default. Moving this to Idea and Feature Request.

**Comment by janis-saltz** [comment]
> Model will always have the higher priority over default. Moving this to Idea and Feature Request 
That seems unintuitive and a bit backwards. 

```php
class SomeModel extends Model {
	protected $attributes = [
		'reason' => 'c',
	];
}

Select::make('Reason', 'reason')
    ->options([
        'a' => 'A`,
        'b' => 'B',
 ])
    ->default('a'),
    
Select::make('Reason 2', 'reason')
    ->options([
        'a' => 'A`,
        'b' => 'B',
 ])
    ->default('b'),
```

Why would a developer expect the provided defaults to be ignored?

Nova documentation states:
> Nova offers this functionality via the default method, which accepts a value or callback. This value will be used as the field's default input value on the resource's creation view:


There's no mention that it might be overridden by the model or anything else. 

When writing the above code, explicitly specifying the default for specific field instances, one would expect the field to default to that value. In this case, both fields would just ignore the provided default value because the model states otherwise and one would have no idea why.

Furthermore, let's say I add the `$attributes` property to a model for nova-unrelated reasons sometime after Nova fields were defined. By doing so, I'd have no idea that I just broke a bunch of fields in across my admin panel. Suddenly, all my `->default()` calls are ignored.

Wouldn't it be a bit more logical to use whatever is explicitly provided and default to what model states in case no explicit default is provided? 
I mean, if I don't explicitly state that "this fields should default to this value", then I expect the field to default to whatever Nova's or the Model's way of determining what the default value should be for the field, not the other way around.
